### ISSUE TITLE
Allow passing in github token as an input

### ISSUE THREAD

**Issue by arturenault** [salesforcecli/github-workflows#22]
I'm here via oclif after using oclif's generator command to generate my CLI:
```
npx oclif generate
```

This generated a bunch of github workflows which depend on oclif's [github workflows](https://github.com/oclif/github-workflows), which are duplicates of this repo. Let me know if I'm opening an issue in the wrong repo.

Unfortunately, none of the workflows work because they all directly reference the secrets the workflows depend on, e.g. [here](https://github.com/salesforcecli/github-workflows/blob/main/.github/workflows/githubRelease.yml#L17).

Since GitHub actions doesn't allow sharing secrets between organizations in different enterprises, this means the token is null and nothing works: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idsecretsinherit

Would be great to have this fixed. I was excited about not having to write my own GitHub actions, but unfortunately this ruins that.

**Comment by cristiand391** [comment]
Yeah, I hit the same errors when trying to publish my plugin outside of the `salesforcecli` org.
We are aware of this and still haven't decided the solution (either publish actions or maintain a "lighter" version for external users.

My solution was to copy the workflows from here and remove some salesforce-specific stuff, with these you can do a manual release from the actions UI and get GH releases -> npm publish done.
https://github.com/cristiand391/sf-plugin-api/tree/main/.github/workflows

**Comment by arturenault** [comment]
I ended up forking the repo myself as well, which worked fine. I had to do this manually using [this method](https://gist.github.com/0xjac/85097472043b697ab57ba1b1c7530274) as github won't let you reference workflows from forks from other orgs, apparently.

```
either publish actions or maintain a "lighter" version for external users
```
You're kind of already doing both, since the actions are published as part of the oclif generator. [They point to the "lighter" version available in the oclif repo.](https://github.com/oclif/github-workflows)
It seems like it'd be pretty easy not to directly reference the secret, and instead accept just accept it as an input parameter. I'm happy to propose a PR if that'd be helpful.
